### PR TITLE
Fixed typo in interface method.

### PR DESCRIPTION
**Pull request by tiagosampaio**
Fixed typo in interface method. The method name was changed from proccessAdditionalValidation() to processAdditionalValidation().

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @tiagosampaio. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Comment by orlangur** [comment]
@tiagosampaio as I mentioned in https://github.com/magento/magento2/pull/19284#issuecomment-440489904, we are trying to understand currently if it is possible to change something like this even in 2.3.

**Comment by tiagosampaio** [comment]
@orlangur sorry. I didn't get it when you said that. Anyway this is a fix that is required to be done once the classes that implement this interface are already changed and using the new method name. The old one needs to be deprecated.

**Comment by orlangur** [comment]
@tiagosampaio the problem is that we cannot add new methods into interface as it will break any existing implementation.

**Comment by tiagosampaio** [comment]
@orlangur any news about this?

File: app/code/Magento/Shipping/Model/Carrier/AbstractCarrierInterface.php [hunk 1]
```diff
@@ -90,7 +90,7 @@ public function checkAvailableShipCountries(\Magento\Framework\DataObject $reque
      * @return $this|\Magento\Framework\DataObject|boolean
      * @api
      */
-    public function proccessAdditionalValidation(\Magento\Framework\DataObject $request);
+    public function processAdditionalValidation(\Magento\Framework\DataObject $request);
```
Comment (sidolov) [inline app/code/Magento/Shipping/Model/Carrier/AbstractCarrierInterface.php:93]: Unfortunately, we are not able to change public methods signature according to our [Backward Compatible Development Guide](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/)

**Review by sidolov — CHANGES_REQUESTED** [review]

**Comment by sivaschenko** [comment]
Closing this pull request due to inactivity, feel free to reopen if you'd like to continue progress

**Comment by orlangur** [comment]
@sidolov @sivaschenko guys, I still didn't receive any answer in Slack of here whether we can apply such kind of changes at all or the only way is to deprecate interface completely and introduce a new one?

**Comment by sivaschenko** [comment]
@orlangur we cannot change/add/remove public and protected methods of API classes (in patch releases)

**Comment by orlangur** [comment]
@sivaschenko I know what we CANNOT do :) Could you please tell me what is an easiest way to fix such typo?

Not in the patch release, but at least in `2.4.0`.

**Comment by sivaschenko** [comment]
@orlangur the only way I know currently is to get an architect approval.

**Comment by orlangur** [comment]
Thanks @sivaschenko!

**Pull request closed without merging**